api/src/main/java/org/pmiops/workbench/db/dao/WorkspaceServiceImpl.java

```diff
@@ -75,6 +80,19 @@ public Workspace getRequired(String ns, String firecloudName) {
     return workspace;
   }
 
+  @Override
+  @Transactional
+  public Workspace getRequiredWithCohorts(String ns, String firecloudName) {
+    Workspace workspace = workspaceDao.findByFirecloudWithEagerCohorts(ns, firecloudName);
+    if (workspace == null) {
+      throw new NotFoundException(String.format("Workspace %s/%s not found.", ns, firecloudName));
+    }
+    // TODO(calbach): Use a NamedEntityGraph rather than this hacky populate approach.
+    // Force an eager load of all cohort reviews while we have a session still open.
+    workspace.getCohorts().stream().forEach(c -> c.getCohortReviews().size());
```

Comment:
Could you instead change findByFirecloudWithEagerCohorts to also fetch reviews?

Reply:
I tried this out with the obvious JOIN w.cohorts c JOIN c.cohortReviews; it failed with:

```
Caused by: org.hibernate.loader.MultipleBagFetchException: cannot simultaneously fetch multiple bags: [org.pmiops.workbench.db.model.Workspace.cohorts, org.pmiops.workbench.db.model.Cohort.cohortReviews]
```

I think it may be more involved to use this approach, I think we may want to use NamedEntityGraphs as suggested by the TODO, but I'm not sure. Let me know if you have any other ideas off the top of your head.

On the upside, it seems that there can only be at most 1 cohort review per cohort (?) at least based on the current code, so this shouldn't be much of a hit.

Reply:
There will normally be one cohort review per cohort, yes... in future there could be multiple (one per CDR version.) I'm not sure whether we want to at that point actually return all of them here or just the one for the CDR version for the workspace.

Anyway for now this is OK.

Reply:
This post:

https://stackoverflow.com/questions/25139879/querying-hierarchical-data-with-jpql-hql

suggests that if we change our collections from List<> to Set<> it may work?

Anyway, I'm fine with it as-is for now.

Reply:
That worked (then caused more problems, which I fixed). I saw that thread but closed it when I saw "cartesian product". Anyways, since a cohort review itself is fairly small (and at most 1 per CDR version), this seems fine. Done.